src/mono/wasm/testassets/WasmBasicTestApp/wwwroot/main.js

```diff
@@ -21,6 +25,29 @@ switch (testCase) {
     case "AppSettingsTest":
         dotnet.withApplicationEnvironment(params.get("applicationEnvironment"));
         break;
+    case "DownloadResourceProgressTest":
+        if (params.get("failFirstAssemblyDownload") === "true") {
+            let hasFailedFetch = false;
+            dotnet.withResourceLoader((type, name, defaultUri, integrity) => {
+                if (hasFailedFetch || type !== "assembly")
+                    return fetch(defaultUri, { integrity: integrity, cache: 'no-cache' });
+
+                testOutput("Throw error instead of downloading resource");
```

Comment:
@radical I have diverged from what we discussed on friday, to push the failing fetch from playwright, because I want the app to serve also as "sample", or to be debuggable by manually running it. Instead I'm checking for this message.

Reply:
Now I'm thinkh it would be better to add a debug log from `assets.ts` when it retries and check that in test.

Reply:
Added a retry message to `asset.ts`